### PR TITLE
UF-7461: Implemented email service for sending information on error

### PR DESCRIPTION
**Pull request by josc04**

### CODE REVIEW

**Review by cheezi0747 — APPROVED** [review]

File: src/main/java/se/sundsvall/billingpreprocessor/service/MessagingService.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+		this.client = client;
+		this.applicationName = applicationName;
+		this.environment = environment;
+		this.properties = properties;
+	}
+
+	public void sendErrorMail(List<CreationError> errors) {
+		if (isBlank(properties.sender()) || isEmpty(properties.recipients())) {
+			return;
```
Comment (foobar-1337) [inline src/main/java/se/sundsvall/billingpreprocessor/service/MessagingService.java:39]: Should we log something here?

**Comment by ronnberg** [comment]
Wrong task number in commit msg

File: src/main/java/se/sundsvall/billingpreprocessor/service/mapper/MessagingMapper.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+		return new EmailSender()
+			.name(sender)
+			.address(sender);
+	}
+
+	private static String base64Encode(String message) {
+		if (StringUtil.isNullOrEmpty(message)) {
+			return message;
+		}
+		try {
```
Comment (ronnberg) [inline src/main/java/se/sundsvall/billingpreprocessor/service/mapper/MessagingMapper.java:90]: use org.apache.commons.codec.binary.Base64.isBase64?
https://mvnrepository.com/artifact/commons-codec/commons-codec/1.16.1

File: src/main/java/se/sundsvall/billingpreprocessor/service/MessagingService.java [hunk 3]
```diff
@@ -17,15 +19,16 @@
...
 
 	private final String applicationName;
 	private final String environment;
 	private final MessagingClient client;
 	private final ErrorMessageProperties properties;
 
 	public MessagingService(MessagingClient client,
-		@Value("${spring.application.name:}") String applicationName,
-		@Value("${spring.profiles.active:default}") String environment,
+		@Value("${spring.application.name:default}") String applicationName,
```
Comment (foobar-1337) [inline src/main/java/se/sundsvall/billingpreprocessor/service/MessagingService.java:30]: Should the application start without an applicationName?

File: src/main/java/se/sundsvall/billingpreprocessor/service/MessagingService.java [hunk 4]
```diff
@@ -17,15 +19,16 @@
 @Service
 public class MessagingService {
 	private static final boolean ASYNCHRONOUSLY = true;
+	private static final Logger LOG = LoggerFactory.getLogger(InvoiceFileService.class);
```
Comment (foobar-1337) [inline src/main/java/se/sundsvall/billingpreprocessor/service/MessagingService.java:22]: Wrong class in logger?

File: src/main/java/se/sundsvall/billingpreprocessor/service/MessagingService.java (hunk 3 above)
Comment (josc04) [reply src/main/java/se/sundsvall/billingpreprocessor/service/MessagingService.java:30]: The application will not start if `spring.application.name` is not present, so the `:` after name is unnecessary - will remove.

**Review by foobar-1337 — APPROVED** [review]

**Pull request merged**